### PR TITLE
Documentation for Fit benchmarking scripts

### LINKED ISSUE #18587: Technical documentation for Fit benchmarking scripts
Tech doc for scripts\CompareFitMinimizers.

Add a .md file as for example is done for scripts\DiamondAttenuationCorrection etc

### PR DESCRIPTION
**Pull request by simonfernandes**
Description of work.

**To test: Whether the readme is accurate and adequate**

Fixes #18587.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/CompareFitMinimizers/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+> > > > > Fitting done via `fitting_benchmarking.run_fit`
+
+> > > > > > Return a list `FittingTestResult` objects per problem file (starting points * minimizers) 
+
+> > > > Return a list of `FittingTestResult` objects for the group
+
+> > Return a nested list of results per group
+
+> The list of groups (or files within) can be passed into a chosen `results_output.py` function to be printed out and optionally saved to file
```
Comment (Anders-Markvardsen) [inline scripts/CompareFitMinimizers/README.md]: Hi @simonfernandes note sure I understand what you mean by "(or files within)" in this sentence. I suggest either delete or reword

File: scripts/CompareFitMinimizers/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+> > > For each file in the group:
+
+> > > > File parsed for relevant data, and a `FittingTestProblem` object is created from this
+
+> > > > Problem object passed into `fitting_benchmarking.do_fitting_benchmark_one_problem`
+
+> > > > > Fitting done via `fitting_benchmarking.run_fit`
+
+> > > > > > Return a list `FittingTestResult` objects per problem file (starting points * minimizers)
```
Comment (Anders-Markvardsen) [inline scripts/CompareFitMinimizers/README.md]: Do you mean to have the indentation one level in from the line above here?

Comment (simonfernandes) [inline scripts/CompareFitMinimizers/README.md]: I did, to indicate that `run_fit` returns that particular list, but this isn't the case actually. I have made some corrections and clarified some wording soon to follow.

**Comment by Anders-Markvardsen** [comment]
Minor comments when creating the pull request copy the labels for the issue over (in this case milestone+fitting+documentation).

Reads well a give a quick useful overview of the script in this folders is doing. :shipit:

No update to release note required

**Comment by martyngigg** [comment]
:+1: 

Just another very minor housekeeping point: the pull request should be assigned to the reviewer and not the author. The reviewer never used to exist and we haven't quite agreed how to use the state yet so we're sticking with the main tester/reviewer assigns themselves the PR.

**Pull request merged**